src/analyze/check_not_modified.cc

```diff
@@ -49,6 +49,10 @@ Stmt InsertTmpEval::visitStmt(const Stmt &_op) {
     return ret;
 }
 
+struct ModifiedException : public Error {
+    ModifiedException() : Error("") {}
+};
```

Comment:
If just used as a control flow rather than a real exception, you don't have to inherit `Error` and there will be less overhead.